index.d.ts

```diff
@@ -7,99 +7,240 @@
-      signatureAlgorithm?: string | undefined;
-    }
-  );
+
+  constructor(idMode?: string | null, options?: SignedXmlOptions);
+
+  /**
+   * Due to key-confusion issues, its risky to have both hmac
+   * and digital signature algos enabled at the same time.
+   * This enables HMAC and disables other signing algos.
+   */
+  enableHMAC(): void;
+
+  /**
+   * Validates the signature of the provided XML document synchronously using the configured key info provider.
+   *
+   * @param xml The XML document containing the signature to be validated.
+   * @returns `true` if the signature is valid
+   * @throws Error if no key info resolver is provided.
+   */
+  checkSignature(xml: string): boolean;
+
+  /**
+   * Validates the signature of the provided XML document asynchronously using the configured key info provider.
+   *
+   * @param xml The XML document containing the signature to be validated.
+   * @param callback Callback function to handle the validation result asynchronously.
+   * @throws Error if the last parameter is provided and is not a function, or if no key info resolver is provided.
+   */
+  checkSignature(xml: string, callback: (error: Error | null, isValid?: boolean) => void): void;
+
+  /**
+   * Loads the signature information from the provided XML node or string.
+   *
+   * @param signatureNode The XML node or string representing the signature.
+   * @throws Error if the canonicalization or signature method elements are not found, or if there are no reference elements.
+   */
+  loadSignature(signatureNode: Node | string): void;
+
+  /**
+   * Adds a reference to the signature.
+   *
+   * @param xpath The XPath expression to select the XML nodes to be referenced.
+   * @param transforms An array of transform algorithms to be applied to the selected nodes. Defaults to ["http://www.w3.org/2001/10/xml-exc-c14n#"].
+   * @param digestAlgorithm The digest algorithm to use for computing the digest value. Defaults to "http://www.w3.org/2000/09/xmldsig#sha1".
+   * @param uri The URI identifier for the reference. If empty, an empty URI will be used.
+   * @param digestValue The expected digest value for the reference.
+   * @param inclusiveNamespacesPrefixList The prefix list for inclusive namespace canonicalization.
+   * @param isEmptyUri Indicates whether the URI is empty. Defaults to `false`.
+   */
   addReference(
     xpath: string,
-    transforms?: ReadonlyArray<string>,
-    digestAlgorithm?: string,
+    transforms?: CanonicalizationAlgorithmType[],
+    digestAlgorithm?: HashAlgorithmType,
     uri?: string,
     digestValue?: string,
     inclusiveNamespacesPrefixList?: string,
     isEmptyUri?: boolean
   ): void;
-  checkSignature(xml: string): boolean;
+
+  /**
+   * Compute the signature of the given XML (using the already defined settings).
+   *
+   * @param xml The XML to compute the signature for.
+   * @returns If no callback is provided, returns `this` (the instance of SignedXml).
```

Comment:
This signature doesn't have an argument for callback, so we don't need to include it in the docblock.